File: src/main/java/com/uitnetwork/alexa/service/AlexaIntentRequestMappingService.java
package com.uitnetwork.alexa.service;

import static com.uitnetwork.alexa.model.AlexaIntent.lookupAlexaIntent;
import static org.slf4j.LoggerFactory.getLogger;

import org.slf4j.Logger;
import org.springframework.stereotype.Service;

import com.amazon.speech.json.SpeechletRequestEnvelope;
import com.amazon.speech.speechlet.IntentRequest;
import com.uitnetwork.alexa.model.AlexaIntent;
import com.uitnetwork.alexa.model.AlexaIntentRequest;

@Service
public class AlexaIntentRequestMappingService {
    private static final Logger logger = getLogger(TimeIntentService.class);

    public AlexaIntentRequest translate(SpeechletRequestEnvelope<IntentRequest> intentRequestEnvelope) {
        logger.info("Mapping for intentName: {}", intentRequestEnvelope.getRequest().getIntent().getName());

        AlexaIntent alexaIntent = lookupAlexaIntent(intentRequestEnvelope.getRequest().getIntent().getName());
        return new AlexaIntentRequest(alexaIntent, intentRequestEnvelope);
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/service/AlexaIntentRequestMappingService.kt
package com.uitnetwork.alexa.service

import com.amazon.speech.json.SpeechletRequestEnvelope
import com.amazon.speech.speechlet.IntentRequest
import com.uitnetwork.alexa.model.AlexaIntent.Companion.lookupAlexaIntent
import com.uitnetwork.alexa.model.AlexaIntentRequest
import org.slf4j.LoggerFactory.getLogger
import org.springframework.stereotype.Service

@Service
class AlexaIntentRequestMappingService {
    private val logger = getLogger(AlexaIntentRequestMappingService::class.java)

    fun translate(intentRequestEnvelope: SpeechletRequestEnvelope<IntentRequest>): AlexaIntentRequest {
        logger.info("Mapping for intentName: {}", intentRequestEnvelope.request.intent.name)

        val alexaIntent = lookupAlexaIntent(intentRequestEnvelope.request.intent.name)
        return AlexaIntentRequest(alexaIntent, intentRequestEnvelope)
    }
}
